# Import Data Set

In [ ]:
import pandas as pd
import seaborn as sns
data = pd.read_csv("/content/concrete_data.csv")
data.head()


Cement  Blast Furnace Slag  Fly Ash  Water  Superplasticizer  \
0   540.0                 0.0      0.0  162.0               2.5   
1   540.0                 0.0      0.0  162.0               2.5   
2   332.5               142.5      0.0  228.0               0.0   
3   332.5               142.5      0.0  228.0               0.0   
4   198.6               132.4      0.0  192.0               0.0   

   Coarse Aggregate  Fine Aggregate  Age  Strength  
0            1040.0           676.0   28     79.99  
1            1055.0           676.0   28     61.89  
2             932.0           594.0  270     40.27  
3             932.0           594.0  365     41.05  
4             978.4           825.5  360     44.30

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1030 entries, 0 to 1029
Data columns (total 9 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Cement              1030 non-null   float64
 1   Blast Furnace Slag  1030 non-null   float64
 2   Fly Ash             1030 non-null   float64
 3   Water               1030 non-null   float64
 4   Superplasticizer    1030 non-null   float64
 5   Coarse Aggregate    1030 non-null   float64
 6   Fine Aggregate      1030 non-null   float64
 7   Age                 1030 non-null   int64  
 8   Strength            1030 non-null   float64
dtypes: float64(8), int64(1)
memory usage: 72.6 KB


# Correlation

In [ ]:
data

Cement  Blast Furnace Slag  Fly Ash  Water  Superplasticizer  \
0      540.0                 0.0      0.0  162.0               2.5   
1      540.0                 0.0      0.0  162.0               2.5   
2      332.5               142.5      0.0  228.0               0.0   
3      332.5               142.5      0.0  228.0               0.0   
4      198.6               132.4      0.0  192.0               0.0   
...      ...                 ...      ...    ...               ...   
1025   276.4               116.0     90.3  179.6               8.9   
1026   322.2                 0.0    115.6  196.0              10.4   
1027   148.5               139.4    108.6  192.7               6.1   
1028   159.1               186.7      0.0  175.6              11.3   
1029   260.9               100.5     78.3  200.6               8.6   

      Coarse Aggregate  Fine Aggregate  Age  Strength  
0               1040.0           676.0   28     79.99  
1               1055.0           676.0   28     61.89  
2                932.0           594.0  270     40.27  
3                932.0           594.0  365     41.05  
4                978.4           825.5  360     44.30  
...                ...             ...  ...       ...  
1025             870.1           768.3   28     44.28  
1026             817.9           813.4   28     31.18  
1027             892.4           780.0   28     23.70  
1028             989.6           788.9   28     32.77  
1029             864.5           761.5   28     32.40  

[1030 rows x 9 columns]

In [ ]:
import statsmodels.formula.api as smf
model = smf.ols('Strength ~ Q("Cement") + Q("Blast Furnace Slag") + Q("Fly Ash") + Water + Superplasticizer + Q("Coarse Aggregate") + Q("Fine Aggregate") + Age', data=data).fit()

In [ ]:
#Coefficients
model.params

Intercept                 -23.331214
Q("Cement")                 0.119804
Q("Blast Furnace Slag")     0.103866
Q("Fly Ash")                0.087934
Water                      -0.149918
Superplasticizer            0.292225
Q("Coarse Aggregate")       0.018086
Q("Fine Aggregate")         0.020190
Age                         0.114222
dtype: float64

In [ ]:
#R squared values
(model.rsquared,model.rsquared_adj)

(0.6155198704142721, 0.6125072934929343)

###
As per R squared value is equal to 61% so strength can be measured 61% accurately by independent factors such as age,
 Cement, Blast Furnace Slag, Fly Ash, Superplasticizer,Coarse Aggregate and Fine Aggregate.


Import Data Set